src/main/java/com/google/sps/data/Guide.java

```diff
@@ -49,4 +50,54 @@ public String getCategory() {
     public long getTimestamp() {
         return timestamp;
     }
+
+    public Error getError(){
+        return error;
+    }
+
+    private boolean emptyString(String str){
+        if(str.equals("")) return true;
+        return false;
+    }
+
+    private boolean moreThan1500bytes(String str){
+        if(title.getBytes().length > 1500) return true;
+        return false;
+    }
+
+    public boolean isInvalid(){
```

Comment:
You've created a stateful API here that I think is somewhat difficult to use, since you have to know to first call isInvalid and then getError. Instead, you could return a status object that has a success or error code. Or, if Error were to extend RuntimeException, you can use a throw statement and change the return type to void. This way, there is no need for saved state.

Reply:
Fixed in next commit